sandy/core/cov.py

```diff
@@ -341,6 +343,100 @@ def corr(self):
                             columns=self.data.columns,
                             )
 
+    def _reduce_size(self):
+        """
+        Reduces the size of the matrix, erasing the null values.
+
+        Returns
+        -------
+        nonzero_idxs : `numpy.ndarray`
+            The indices of the diagonal that are not null.
+        cov_reduced : `pandas.DataFrame`
+            The reduced matrix.
+
+        Examples
+        --------
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([0, 1, 2], dtype=int64)
+        >>> reduce_matrix
+                      0	          1	          2
+        0	1.00000e+00	0.00000e+00	0.00000e+00
+        1	0.00000e+00	2.00000e+00	0.00000e+00
+        2	0.00000e+00	0.00000e+00	3.00000e+00
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([0, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([1, 2], dtype=int64)
+        >>> reduce_matrix
+                      1	          2
+        1	2.00000e+00	0.00000e+00
+        2	0.00000e+00	3.00000e+00
+        """
+        nonzero_idxs = np.flatnonzero(np.diag(self.data))
+        cov_reduced = self.data.loc[nonzero_idxs, nonzero_idxs]
+        return nonzero_idxs, cov_reduced
+
+    def _restore_size(cls, nonzero_idxs, cov_reduced, dim):

```

Comment:
why `cls` and not `self`?

Reply:
you don't use `self`, then you can remove it and make it a static method.
But, as reported below, maybe it should be a class method